lint/ktlint.bzl

```diff
@@ -0,0 +1,176 @@
+in your WORKSPACE or MODULE.bazel. Then create a `ktlint` binary target to be used in your linter as follows, typically in `tools/linters/BUILD.bazel`:
+
+```
+java_binary(
+    name = "ktlint",
+    main_class = "com.pinterest.ktlint.Main",
+    runtime_deps = [
+        "@maven//:com_pinterest_ktlint_ktlint_cli",
+    ],
+)
+```
+
+```
+
+Then, create the linter aspect, typically in `tools/lint/linters.bzl`:
+
+```starlark
+load("@aspect_rules_lint//lint:ktlint.bzl", "ktlint_aspect")
+
+ktlint = ktlint_aspect(
+    binary = "@@//tools/linters:ktlint",
+    # rules can be enabled/disabled from with this file
+    editorconfig = "@@//:.editorconfig",
+    # a baseline file with exceptions for violations
+    baseline_file = "@@//:.ktlint-baseline.xml",
+)
+```
+"""
+
+load("//lint/private:lint_aspect.bzl", "LintOptionsInfo", "filter_srcs", "report_file")
+
+_MNEMONIC = "ktlint"
+
+def ktlint_action(ctx, executable, srcs, editorconfig, report, baseline_file, use_exit_code = False):
+    """ Runs ktlint as build action in Bazel.
+
+    Adapter for wrapping Bazel around
+    https://pinterest.github.io/ktlint/latest/install/cli/
+
+    Args:
+        ctx: an action context or aspect context
+        executable: struct with ktlint field
+        srcs: A list of source files to lint
+        editorconfig: The file object pointing to the editorconfig file used by ktlint
+        report: :output:  the stdout of ktlint containing any violations found
+        baseline_file: The file object pointing to the baseline file used by ktlint.
+        use_exit_code: whether a non-zero exit code from ktlint process will result in a build failure.
+    """
+
+    args = ctx.actions.args()
+    inputs = srcs
+    outputs = [report]
+
+    if not use_exit_code:
+        args.add(executable.path)
+
+    if editorconfig:
+        inputs.append(editorconfig)
+        args.add("--editorconfig={}".format(editorconfig.path))
+    if baseline_file:
+        inputs.append(baseline_file)
+        args.add("--baseline={}".format(baseline_file.path))
+
+    args.add("--relative")
+
+    if use_exit_code:
+        args.add("--reporter=plain,output={}".format(report.path))
+        ctx.actions.run(
+            inputs = inputs,
+            outputs = outputs,
+            executable = executable,
+            arguments = [args],
+            mnemonic = _MNEMONIC,
+        )
+    else:
+        args.add("--reporter=plain,output={}".format(report.path))
+        ctx.actions.run_shell(
+            inputs = inputs,
+            outputs = outputs,
+            command = """
+            ktlint=$1
+            ktlint_args="${@:2}"
+            # Don't fail ktlint and just report the violations
+            $ktlint ${ktlint_args} || true
+            """,
+            mnemonic = _MNEMONIC,
+            arguments = [args],
+            tools = [executable],
+        )
+
+def _ktlint_aspect_impl(target, ctx):
+    if ctx.rule.kind not in ["kt_jvm_library", "kt_jvm_binary"]:
```

Comment:
what about the other `kt` rules that target non-jvm?

Reply:
Updated to include `kt_js_library` which i think is the only non-jvm rule that uses `srcs`.